### ISSUE TITLE
Support clang on Linux

### ISSUE THREAD

**Issue by bverhagen** [aphenriques/integral#11]
Currently, clang is supported on MAC OS X, however, it is also getting a rather popular compiler on Linux systems.

At the moment, the 'decision' on whether the compiler is the GCC or Clang compiler is made based on determining the system that is used.

Given that GCC and Clang are rather similar, this shouldn't be too hard. Currently, the only 'problem' is:
```
$ make CXX=clang++
error: unknown warning option '-Wno-unused-but-set-parameter'; did you mean '-Wno-unused-parameter'? [-Werror,-Wunknown-warning-option].
```

Does @aphenriques prefer to:
- Detect GCC vs Clang in the Makefile
- Fix this warning in the code and remove this warning suppression?
```

**Comment by bverhagen** [comment]
Given that the OS is already checked, I went for the first option (detecting GCC vs Clang). It may be useful in the future for other compiler options.

It may be worth fixing the `unused-but-set-parameter` nevertheless, although I do not immediately see a good fix without digging in further.

**Comment by aphenriques** [comment]
Thanks again for this report @bverhagen !

I ended up removing all the warning suppression flags on d307f9a4c3d38c2ac9ec5efe9a0813e9cf29b27c. I left some unused function parameters as a documentation mechanism for me, but that was not a good idea. The remaining OS specific definitions on common.mk are appropriate. The resulting common.mk is simpler, which is good.

Sorry I pushed the changes after your pull request. I will close your pull request in favor of d307f9a4c3d38c2ac9ec5efe9a0813e9cf29b27c. That was a nice trick though! I am glad I learned something new.
